Storage/REST-API.md

```diff
@@ -0,0 +1,185 @@
+# RESTful binding for Storage, RFC
+
+This document describes the RESTful API with CRUD functionality for Current Storage.
+
+### Bindings semantics
+`TODO: Describe storage field <-> endpoint bindings.`
+
+### Response schema
+
+The response body is always either a valid JSON or empty with `204 (No content)` error code.
+
+The generic format of a successful JSON response is:
+
+```
+{
+  "url": "https://api.example.com/...",
+  "url_*": "https://api.example.com/...",
+  ... more "url_*" fields ...
+  
+  [optional]
+  "meta": {
+    ...
+  },
+  
+  [required]
+  "data": {
+    ...
+  }
+
+}
+```
+
+The responses are strongly typed. The contents of the request URL define the schema for the response.
+
+The value for `"data"` can be:
+* an object,
+* an array, if the response is expected to contain a list,
+* primitive type (string, number, boolean) in special cases.
+
+Error responses are JSON objects with the top-level `"error":...` field:
+
+```
+{
+  “error”: “Detailed human-readable error message.”,
+  ... other fields ...
+}
+```
+
+Successful responses are JSON objects without the top-level `"error"` field.
+
+`TODO: We also may have redirects, which contain the "Location" in the header, with empty body.`
```

Comment:
Technically, 3xx responses and successes. There's no need to mentioned them at all, as this is implied by REST as a part of the unified HTTP interface. In practical terms, I've never seen a library that wouldn't handle redirects by default (my experience is not exhaustive, of course.)